php/Blockonomics.php

```diff
@@ -131,56 +131,34 @@ public function check_get_callbacks_response_code($response){
     public function check_get_callbacks_response_body ($response, $crypto){
         $error_str = '';
         $response_body = json_decode(wp_remote_retrieve_body($response));
-
-        $callback_secret = get_option('blockonomics_callback_secret');
-        $api_url = WC()->api_request_url('WC_Gateway_Blockonomics');
-        $callback_url = add_query_arg('secret', $callback_secret, $api_url);
-        $callback_url_without_schema = preg_replace('/https?:\/\//', '', $callback_url);
-
         if (!isset($response_body) || count($response_body) == 0)
         {
             $error_str = __('You have not entered an xPub', 'blockonomics-bitcoin-payments');
         }
         elseif (count($response_body) == 1)
         {
             $response_callback = '';
-            $response_address = '';
+            $xpub = '';
             if(isset($response_body[0])){
                 $response_callback = isset($response_body[0]->callback) ? $response_body[0]->callback : '';
-                $response_address = isset($response_body[0]->address) ? $response_body[0]->address : '';
+                $xpub = isset($response_body[0]->address) ? $response_body[0]->address : '';
             }
-            $response_callback_without_schema = preg_replace('/https?:\/\//', '', $response_callback);
-            if(!$response_callback || $response_callback == null)
-            {
-                //No callback URL set, set one 
-                $this->update_callback($callback_url, $crypto, $response_address);
```

Comment:
This code automatically sets the callback url if the xpub has an empty address callback url and is missing from the new logic.

Reply:
True, thanks. It's now been fixed.